src/main/scala/com/atomist/source/git/GitHubServices.scala

```diff
@@ -55,29 +55,31 @@ case class GitHubServices(oAuthToken: String, apiUrl: Option[String] = None)
                               body: Option[Array[Byte]] = None,
                               params: Map[String, String] = Map.empty,
                               headers: Map[String, String] = headers)(implicit m: Manifest[T]): T =
-    (body match {
-      case Some(data) => method match {
-        case Post => Http(url).postData(data)
-        case Put => Http(url).put(data)
-        case Patch => Http(url).postData(data).method(method.toString)
-        case _ => Http(url).postData(data)
-      }
-      case None => Http(url).method(method.toString)
-    }).headers(headers)
-      .params(params)
-      .exec((code: Int, headers: Map[String, IndexedSeq[String]], is: InputStream) => code match {
-        case 200 | 201 => fromJson[T](is)
-        case success if 202 until 206 contains success => ().asInstanceOf[T]
-        case 304 => ().asInstanceOf[T]
-        case 401 | 403 | 415 | 422 =>
-          val msg = IOUtils.toString(is, defaultCharset)
-          logger.warn(s"${headers("Status").head}, $msg")
-          throw DoNotRetryException(msg)
-        case _ =>
-          val msg = IOUtils.toString(is, defaultCharset)
-          logger.warn(s"${headers("Status").head}, $msg")
-          throw ArtifactSourceException(msg)
-      }).body
+    retry("httpRequest") {
```

Comment:
Agreed. I did the same for retrying in `paginateResults` and `paginateSearchResults`